[incubator/jaeger] Adds ability to utilize a cloud providers firewall for services

jtslear opened the pull request:
* Adds declaration allowing the user to configure which IP CIDRs are
allowed to access a service via a cloud providers firewall
* An example configuration of `values.yaml` may look like this:
```
query:
  service:
    type: LoadBalancer
    loadBalancerSourceRanges:
      - "192.0.2.0/24"
      - "198.51.100.0/24"
```
* The above would configure a cloud provider's firewall with a ruleset
allowing only the configured IP ranges access to the query service
* This adds this declaration across the board for all services as it may
be valuable. One example, is to send metric data from applications that
do not reside in kubernetes
* This PR also updates the README to reflect other options for services
that have not been updated over time.

jtslear committed:
Adds ability to utilize a cloud providers firewall for services

* Adds declaration allowing the user to configure which IP CIDRs are
allowed to access a service via a cloud providers firewall
* An example configuration of `values.yaml` may look like this:
```
query:
  service:
    type: LoadBalancer
    loadBalancerSourceRanges:
      - "192.0.2.0/24"
      - "198.51.100.0/24"
```
* The above would configure a cloud provider's firewall with a ruleset
allowing only the configured IP ranges access to the query service
* This adds this declaration across the board for all services as it may
be valuable. One example, is to send metric data from applications that
do not reside in kubernetes
* This PR also updates the README to reflect other options for services
that have not been updated over time.

jtslear commented:
/assign @mgoodness

unguiculus commented:
/ok-to-test

pavelnikolov commented on incubator/jaeger/templates/agent-svc.yaml:
```diff
@@ -38,4 +38,10 @@ spec:
     component: agent
     release: {{ .Release.Name }}
     jaeger-infra: agent-instance
+{{- if .Values.agent.service.loadBalancerSourceRanges }}
```
This block is repeated four times, exactly the same. Please, define a template and reuse it instead.

pavelnikolov commented on incubator/jaeger/values.yaml:
```diff
@@ -86,6 +86,9 @@ agent:
     useHostPort: false
   service:
     annotations: {}
+
```
Please, remove empty line

pavelnikolov commented on incubator/jaeger/values.yaml:
```diff
@@ -118,6 +121,9 @@ collector:
   replicaCount: 1
   service:
     annotations: {}
+
```
Please, remove empty line

pavelnikolov commented on incubator/jaeger/values.yaml:
```diff
@@ -150,6 +156,9 @@ query:
   service:
     annotations: {}
     type: ClusterIP
+
```
Please, remove empty line

pavelnikolov commented on incubator/jaeger/values.yaml:
```diff
@@ -211,6 +220,9 @@ hotrod:
     annotations: {}
     name: hotrod
     type: ClusterIP
+
```
Please, remove empty line

pavelnikolov reviewed (changes requested):
Please define a reusable template for the `loadBalancerSourceRanges`.

jtslear committed:
rids empty lines

jtslear committed:
Utilizes a template for the lb source ranges, passes in appropriate
scope to allow each service to be configured individually if required by
administrator

jtslear committed:
missed an empty line

jtslear commented:
I've addressed the comments and performed a rebase.  Please enjoy.

unguiculus commented:
ping @pavelnikolov

unguiculus commented:
/assign

pavelnikolov commented:
lgtm

unguiculus commented:
/lgtm
